### PR TITLE
Download CLI Update script from CLI repo

### PR DESCRIPTION
**Pull request by wtgodbe**
CC @gkhanna79 @brthor @eerhardt

### CODE REVIEW

**Comment by brthor** [comment]
:shipit:  when green

**Comment by gkhanna79** [comment]
I think the build.PS1 file also needs the fix.

**Comment by wtgodbe** [comment]
@gkhanna79 looks like the .PS1 script already downloads dotnet-install from CLI - https://github.com/dotnet/core-setup/blob/release/1.0.0/build_projects/dotnet-host-build/build.ps1#L81

However for some reason, my cherry-pick here didn't pick up the deletion of the dotnet-install.sh script in core-setup, so I'll add that change & rebase.

**Comment by wtgodbe** [comment]
@brthor am I using the wrong channel/CLI branch here? https://github.com/dotnet/core-setup/pull/413/files#diff-7b84801f06d5633f2cf51f06931e27c8R96

File: build_projects/dotnet-host-build/build.sh [hunk 1]
```diff
@@ -93,8 +93,8 @@ done < "$REPOROOT/branchinfo.txt"
 [ -z "$DOTNET_INSTALL_DIR" ] && export DOTNET_INSTALL_DIR=$REPOROOT/.dotnet_stage0/$(uname)
 [ -d "$DOTNET_INSTALL_DIR" ] || mkdir -p $DOTNET_INSTALL_DIR
 
-
-$REPOROOT/scripts/obtain/dotnet-install.sh --channel preview
+DOTNET_INSTALL_SCRIPT_URL="https://raw.githubusercontent.com/dotnet/cli/rel/1.0.0/scripts/obtain/dotnet-install.sh"
+curl -sSL "$DOTNET_INSTALL_SCRIPT_URL" | bash /dev/stdin --channel rel-1.0.0 --verbose
```
Comment (eerhardt) [inline build_projects/dotnet-host-build/build.sh]: We should be keeping the non-Windows and the Windows scripts in line as much as possible.  We shouldn't get out of sync between what windows does and what non-Windows does.

https://github.com/dotnet/core-setup/blob/release/1.1.0/build_projects/dotnet-host-build/build.ps1#L104

File: build_projects/dotnet-host-build/build.sh [hunk 2]
```diff
@@ -93,8 +93,8 @@ done < "$REPOROOT/branchinfo.txt"
 [ -z "$DOTNET_INSTALL_DIR" ] && export DOTNET_INSTALL_DIR=$REPOROOT/.dotnet_stage0/$(uname)
 [ -d "$DOTNET_INSTALL_DIR" ] || mkdir -p $DOTNET_INSTALL_DIR
 
-
-$REPOROOT/scripts/obtain/dotnet-install.sh --channel preview
+DOTNET_INSTALL_SCRIPT_URL="https://raw.githubusercontent.com/dotnet/cli/rel/1.0.0/scripts/obtain/dotnet-install.sh"
+curl -sSL "$DOTNET_INSTALL_SCRIPT_URL" | bash /dev/stdin --verbose
```
Comment (gkhanna79) [inline build_projects/dotnet-host-build/build.sh]: Why are we not specifying the stable version here to be used?

File: test/HostActivationTests/project.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
   "dependencies": {
     "Microsoft.NETCore.App": {
       "type": "platform",
-      "version": "1.0.0"
+      "version": "1.0.2"
```
Comment (wtgodbe) [inline test/HostActivationTests/project.json]: I was seeing the following errors in the Windows builds:

11:53:28 The specified framework 'Microsoft.NETCore.App', version '1.0.0' was not found.
11:53:28   - Check application dependencies and target a framework version installed at:
11:53:28       D:\j\workspace\release_windo---d8901c9b\.dotnet_stage0\Windows\x64\shared\Microsoft.NETCore.App
11:53:28   - The following versions are installed:
11:53:28       1.0.2-servicing-004528-00
11:53:28   - Alternatively, install the framework version '1.0.0'. 

@brthor suggested bumping the version of that package, he can probably tell you more. I'm still tinkering.

Comment (gkhanna79) [inline test/HostActivationTests/project.json]: If we are not buildign CLI for any new platforms than what is supported in 1.0.0, we dont need to change the version (you will still see break with the version update, all the same) but specify the RTM version of CLI as I mentioned in a separate comment.

File: build_projects/dotnet-host-build/build.sh (hunk 2 above)
Comment (wtgodbe) [inline build_projects/dotnet-host-build/build.sh]: @brthor suggested using the default channel (which I believe is the RTM version you mentioned), which means not passing a 'channel' arg. This appears to have worked for non-windows, I'm trying the same for Windows.

File: test/HostActivationTests/project.json (hunk 3 above)
Comment (wtgodbe) [inline test/HostActivationTests/project.json]: @gkhanna79 is "1.0.0-preview2-003133" exactly the string to pass for the '--channel' arg ('--channel 1.0.0-preview2-003133')? Previously we were using 'rel-1.0.0' which seems to suggest a different string format

File: build_projects/dotnet-host-build/build.sh (hunk 2 above)
Comment (wtgodbe) [inline build_projects/dotnet-host-build/build.sh]: @gkhanna79 so I should pass '--version 1.0.0-preview2-003133' here (for both scripts)?

**Comment by brthor** [comment]
Your shell script is pulling the install script from the rel/1.0.0 branch:
`https://raw.githubusercontent.com/dotnet/cli/rel/1.0.0/scripts/obtain/dotnet-install.sh`

But the ps1 on this branch is pulling rel/1.0.0-preview2

On this branch you should pull the script from rel/1.0.0-preview2 so you get the latest preview2 bits by default. There were breaking changes between rel/1.0.0-preview2 and rel/1.0.0 install scripts

**Comment by wtgodbe** [comment]
@brthor the rel/1.0.0-preview2 branch is causing failures using both the 'preview' and 'rel-1.0.0' channels, any ideas why the non-preview2 branch was working?

**Comment by brthor** [comment]
The test project needs to be updated to the servicing version it looks like

```
The specified framework 'Microsoft.NETCore.App', version '1.0.0' was not found.
  - Check application dependencies and target a framework version installed at:
      /Users/dotnet-bot/j/workspace/dotnet_core-setup/release_1.0.0/release_osx_x64_prtest/.dotnet_stage0/Darwin/shared/Microsoft.NETCore.App
  - The following versions are installed:
      1.0.2-servicing-004528-00
  - Alternatively, install the framework version '1.0.0'.
```

**Comment by wtgodbe** [comment]
We are green! I'm gonna rebase then merge.

**Comment by wtgodbe** [comment]
Rebase is being weird in this branch since I cherry-picked changes from Master, so I reopened an identical PR w/ 1 commit here - https://github.com/dotnet/core-setup/pull/415. Closing this.

@gkhanna79 @eerhardt @brthor

**Pull request closed without merging**